src/librustdoc/markdown.rs

```diff
@@ -135,6 +135,9 @@ crate fn test(mut options: Options) -> Result<(), String> {
     find_testable_code(&input_str, &mut collector, codes, options.enable_per_target_ignores, None);
 
     options.test_args.insert(0, "rustdoctest".to_string());
+    if options.nocapture {
+        options.test_args.push("--nocapture".to_string());
+    }
```

Comment:
Why is this duplicated between here and `src/librustdoc/doctest.rs`?

Reply:
They both need to set the option on `libtest` too.

Reply:
But why are there two different places running doctests? What's the difference between this one and `doctest.rs`?

Reply:
Ah right sorry. So one is when you pass a markdown document as input, the other is while we go through codeblocks we found in the doc comments.

Reply:
Hmm, so `markdown.rs` is for `rustdoc foo.md` and `doctest.rs` is for `rustdoc foo.rs`? It shouldn't happen in this PR, but we should deduplicate this logic. Can you open an issue for that?

Reply:
Absolutely, opening issue.